src/components/MultiSelect/MultiSelect.tsx

```diff
@@ -1,138 +1,433 @@
+        }
+        // Return the updated isDisabled state for the parent item
+        return item.isDisabled;
       };
 
-      if (type === "listbox") {
-        const listboxOnChange = onChange as ListboxOnChange;
+      const onChangeSearch = (event) => {
+        const value = event.target.value.trim().toLowerCase();
 
+        if (value === "") {
+          return listItemsCount === defaultItemsVisible
+            ? displayDefaultItems()
+            : setItemsList(items);
+        }
+
+        const filteredItems = items.filter((item) => {
+          if (item.children) {
+            if (item.name.toLowerCase().includes(value)) {
+              return true;
+            }
+
+            return item.children.some((childItem) =>
+              childItem.name.toLowerCase().includes(value)
+            );
+          }
+
+          return item.name.toLowerCase().includes(value);
+        });
+
+        setItemsList(filteredItems);
+      };
+
+      const clearSearchKeyword = () => {
+        if (listOverflow === "scroll") {
+          setItemsList(items);
+        } else {
+          listItemsCount === defaultItemsVisible
+            ? displayDefaultItems()
+            : setItemsList(items);
+        }
+      };
+
+      const displayDefaultItems = () => {
+        const list = [];
+        let count = 0;
+
+        for (let i = 0; i < items.length && count < defaultItemsVisible; i++) {
+          const currentItem = { ...items[i] };
+          count++;
+          if (currentItem.children) {
+            const showChilds = isAllChecked(id, currentItem);
+            const showIndeterminateChilds = isIndeterminate(id, currentItem);
+            count +=
+              !showChilds || !showIndeterminateChilds
+                ? currentItem.children?.length
+                : items[i].children.length;
+          }
+
+          list.push(currentItem);
+          if (count >= defaultItemsVisible) {
+            break;
+          }
+        }
+        setItemsList(list);
+      };
+
+      const showSearchInputBox = () => {
+        if (isSearchable) {
+          return (
+            <TextInput
+              id="multi-select-text-input-id"
+              labelText={`Search ${buttonText}`}
+              isClearable={true}
+              isClearableCallback={clearSearchKeyword}
+              isRequired
+              placeholder={`Search ${buttonText}`}
+              onChange={onChangeSearch}
+              showLabel={false}
+              showRequiredLabel={false}
+              step={1}
+              textInputType="default"
+              type="text"
+              __css={styles.menuSearchInputBox}
+              marginBottom="s"
+            />
+          );
+        }
+      };
+
+      const displayAccordionData = () => {
         return (
-          <MultiSelectListBox
-            {...commonProps}
-            ref={ref}
-            onChange={listboxOnChange}
-            {...rest}
-          />
+          <>
```

Comment:
You don't need this `<>` wrapper because the `Box` component is the only top level component.